### ISSUE TITLE
Mixture of multivariate distributions

### ISSUE THREAD

**Issue by dustinvtran** [tensorflow/tensorflow#11309]
Consider a mixture of 2-dimensional Gaussians in TensorFlow:
```python
ds = tf.contrib.distributions

cat = ds.Categorical(probs=[0.3, 0.2, 0.5])
comps = [ds.MultivariateNormalDiag(loc=[-5.0, -5.0], scale_diag=tf.ones(2)),
         ds.MultivariateNormalDiag(loc=[0.0, 0.0], scale_diag=tf.ones(2)),
         ds.MultivariateNormalDiag(loc=[5.0, 5.0], scale_diag=tf.ones(2))]

mix = ds.Mixture(cat=cat, components=comps)
```
This works because each `MultivariateNormalDiag` distribution has a batch shape of `()`. It is compatible with the `Categorical`'s batch shape.

Now consider a mixture of 2-dimensional Bernoulli's, or Gamma's, or Laplace, or StudentT's. We need an equivalent `MultivariateBernoulli`, `MultivariateGamma`, etc. distribution which allows us to fix the batch shape and increase the event shape.

Are there plans to make such distributions available? 

What about edge cases such as a matrixvariate (k-variate) Bernoulli, where additional parameter dimensions determine the batch shape and the event shape is fixed at 2 (k)?

@ebrevdo , @jvdillon. Issue motivated by https://github.com/blei-lab/edward/issues/686.

**Comment by andydavis1** [comment]
@ebrevdo can you comment on this if you have a chance?

**Comment by dustinvtran** [comment]
@jvdillon Do you think `ds.Independent` is the answer to this problem?

**Comment by jvdillon** [comment]
If youre ok with all elements being independent, then yes I believe `Independent` is what you want.  If you'd like to induce correlation, then you can, eg, used `TransformedDistribution` with the `Affine` bijector.

**Comment by ahq1993** [comment]
Hi,
I am facing this issue:
ds=tf.contrib.distributions
#input is computed by passing some place_holders for tensors through neural networks
sm_scores=tf.nn.softmax(input)#input=batch_sizexseq
cat=ds.Categorical(probs=sm_scores)
mean=nmean x batchx dimens
comps = [ds.MultivariateNormalDiag(loc=[-5.0, -5.0], scale_diag=tf.ones(2)),
         ds.MultivariateNormalDiag(loc=[0.0, 0.0], scale_diag=tf.ones(2)),
         ds.MultivariateNormalDiag(loc=[5.0, 5.0], scale_diag=tf.ones(2))]

**Comment by arainboldt** [comment]
@jvdillon I'd like to implement a `MultivariateBernoulli` with induced correlations as mentioned in your last comment. I'm not too familiar with the `tfp`, but I've found an example from [here](https://arxiv.org/pdf/1711.10604.pdf):


```
vector_laplace = tfd.TransformedDistribution(
                                              distribution=tfd.Laplace(loc=0., scale=1.),
                                              bijector=tfb.Affine(
                                                                    shift=tf.Variable(tf.zeros(d)),
                                                                    scale_tril=tfd.fill_triangular(
                                                                                             tf.Variable(tf.ones(d*(d+1)/2))
                                                                    )
                                              ),
                                              event_shape=[d]
)
```

Unfortunately, the `tfb.Affine` class seems to have been deprecated. Is this just composed by chaining `tfb.Shift` and `tfb.Scale`. If not, how would this be replicated with the current TFP API?

Also, the `tfb.Scale` class doesn't take a lower triangle covariance matrix for the scale parameter, does this change the parameterization and optimization of this distribution?

Thanks!
